validator/accounts/wallet_edit.go

```diff
@@ -21,36 +21,32 @@ func EditWalletConfigurationCli(cliCtx *cli.Context) error {
 	if err != nil {
 		return errors.Wrap(err, "could not open wallet")
 	}
-	switch w.KeymanagerKind() {
-	case keymanager.Local:
-		return errors.New("not possible to edit local keymanager configuration")
-	case keymanager.Derived:
-		return errors.New("derived keymanager is not yet supported")
-	case keymanager.Remote:
-		enc, err := w.ReadKeymanagerConfigFromDisk(cliCtx.Context)
-		if err != nil {
-			return errors.Wrap(err, "could not read config")
-		}
-		opts, err := remote.UnmarshalOptionsFile(enc)
-		if err != nil {
-			return errors.Wrap(err, "could not unmarshal config")
-		}
-		log.Info("Current configuration")
-		// Prints the current configuration to stdout.
-		fmt.Println(opts)
-		newCfg, err := userprompt.InputRemoteKeymanagerConfig(cliCtx)
-		if err != nil {
-			return errors.Wrap(err, "could not get keymanager config")
-		}
-		encodedCfg, err := remote.MarshalOptionsFile(cliCtx.Context, newCfg)
-		if err != nil {
-			return errors.Wrap(err, "could not marshal config file")
-		}
-		if err := w.WriteKeymanagerConfigToDisk(cliCtx.Context, encodedCfg); err != nil {
-			return errors.Wrap(err, "could not write config to disk")
-		}
-	default:
-		return fmt.Errorf(errKeymanagerNotSupported, w.KeymanagerKind())
+	if w.KeymanagerKind() != keymanager.Remote {
```

Comment:
also a great change. but still section still being specific to remote, i wonder if it can be reorganized to be more together for someone to catch this much easier reading through the code. Maybe we can move those functions into a separate wrapper function just to break it up a little.

Reply:
I moved the actual editing logic into a non-exported function called `editRemoteWallet` to improve readability a bit. Nothing else jumped out at me as cleaner. I think if more Keymanager types start supporting wallet configuration edits in the future then there will be room to move this logic into a keymanager interface function, but if we did that now it would only be implemented on the remote keymanager and just return errors for the other types which doesn't see like a good tradeoff to me. WDYT?